System.IO.Ports/SerialPort.cs

```diff
@@ -383,14 +383,14 @@ public string NewLine
         /// <exception cref="IOException">The port is in an invalid state. -or- An attempt to set the state of the underlying
         /// port failed. For example, the parameters passed from this <see cref="SerialPort"/>
         /// object were invalid.</exception>
-        /// <exception cref="ArgumentOutOfRangeException">The read time-out value is less than zero and not equal to <see cref="Timeout"/>.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">The read time-out value is less than zero and not equal to <see cref="Timeout.Infinite"/>.</exception>
         public int ReadTimeout
         {
             get => _readTimeout;
 
             set
             {
-                if (value <= 0)
+                if (value is < 0 && value != Timeout.Infinite)
```

Comment:
I am not particularly a fan of being a more "functional" style like python (like to see a corolation between C and C#), but if we are doing this, then why not use `if (value is < 0 and value is not Timeout.Infinite)`

Reply:
Plus, what happens on zero? should that be set to the "default" value?

Reply:
Zero will be zero, the native driver handles it. No need to do anything here.

Reply:
I would be inclined to make that a comment...